crates/pallet-executor/src/lib.rs

```diff
@@ -371,22 +376,30 @@ mod pallet {
                         );
                         return InvalidTransactionCode::Bundle.into();
                     }
-                    let primary_number = signed_opaque_bundle.bundle.receipt.primary_number;
+
+                    let first_primary_number = signed_opaque_bundle
+                        .bundle
+                        .receipts
+                        .get(0)
+                        .expect("Receipts in a bundle must be non-empty; qed")
+                        .primary_number;
 
                     let builder = ValidTransaction::with_tag_prefix("SubspaceSubmitBundle")
                         .priority(TransactionPriority::MAX)
-                        .and_provides(primary_number)
+                        .and_provides(first_primary_number)
                         .longevity(TransactionLongevity::MAX)
                         .propagate(true);
 
                     // primary_number is ensured to be larger than the best execution chain chain
                     // number above.
                     //
                     // No requires if it's the next expected execution chain number.
```

Comment:
No requires means it can be put into the ready queue directly once validated, otherwise it will be put into the future queue in the tx pool. If it's the next expected one, then it should be just in the ready queue not in the future queue waiting for the missing tag to be satisfied.

Reply:
Wouldn't the condition be satisfied immediately if you did actually have previous ER though?

Reply:
Nope. The tags matching are entirely in the tx pool, it won't check if the previous ER is already included on chain. If you add a `requires` here, it would just wait until an extrinsic providing this ER is received, but it will never be satisfied as that extrinsic has been consumed and the tx pool won't see it again.